gobblin-metastore/src/main/java/gobblin/metastore/util/StateStoreCleaner.java

```diff
@@ -85,17 +89,30 @@ public StateStoreCleaner(Properties properties) throws IOException {
 
   /**
    * Run the cleaner.
+   * @throws ExecutionException
    */
-  public void run() throws IOException {
+  public void run() throws IOException, ExecutionException {
     FileStatus[] stateStoreDirs = this.fs.listStatus(this.stateStoreRootDir);
     if (stateStoreDirs == null || stateStoreDirs.length == 0) {
       LOGGER.warn("The state store root directory does not exist or is empty");
       return;
     }
 
+    List<Future<?>> futures = Lists.newArrayList();
+
     for (FileStatus stateStoreDir : stateStoreDirs) {
-      this.cleanerRunnerExecutor.submit(
-          new CleanerRunner(this.fs, stateStoreDir.getPath(), this.retention, this.retentionTimeUnit));
+      futures.add(this.cleanerRunnerExecutor.submit(new CleanerRunner(this.fs, stateStoreDir.getPath(), this.retention,
+          this.retentionTimeUnit)));
+    }
+
+    ExecutorsUtils.shutdownExecutorService(cleanerRunnerExecutor, Optional.of(LOGGER), 60, TimeUnit.SECONDS);
```

Comment:
Usually you would put this call after checking that all tasks have been completed (e.g. switch this with lines 110-115).